opentracing/src/main/java/opentracing/Tracer.java

```diff
@@ -55,5 +54,23 @@
    * (Note that some OpenTracing implementations consider the Spans on either side of an RPC to have the same identity,
    * and others consider the caller to be the parent and the receiver to be the child)
    */
-  <T> Span join(String operationName, T carrier);
+  <T> SpanBuilder join(String operationName, T carrier);
```

Comment:
> So an assertion won't work, or at least not that particular assertion.

Yeah i picked up on that after I wrote the comment. See the lastest code.
Mandatory arguments are used when first obtaining a SpanBuilder, ie tracer.buildSpan(name) and tracer.join(carrier).

Otherwise the builder can throw exceptions is the same setter is called twice.